File: app/src/main/java/com/simon/harmonichackernews/linkpreview/LinkPreviewJsonUtils.java
package com.simon.harmonichackernews.linkpreview;

import android.text.TextUtils;

import org.json.JSONObject;

final class LinkPreviewJsonUtils {

    private LinkPreviewJsonUtils() {
    }

    static String getString(JSONObject jsonObject, String key) {
        String input = jsonObject.optString(key);
        if (TextUtils.isEmpty(input) || "null".equals(input)) {
            return null;
        }
        return input;
    }
}

File: app/src/main/java/com/simon/harmonichackernews/linkpreview/LinkPreviewJsonUtils.kt
package com.simon.harmonichackernews.linkpreview

import com.simon.harmonichackernews.serialization.JsonObject as JSONObject

internal object LinkPreviewJsonUtils {
    fun getString(jsonObject: JSONObject, key: String): String? =
        jsonObject.optString(key).takeUnless { it.isEmpty() || it == "null" }
}
